### ISSUE TITLE
split with hyperlinks

### ISSUE THREAD

**Issue by chrisrex** [qpdf/qpdf#1003]
Hi,
I want to split a multi-page PDF into chapters. There are hyperlinks within the chapters.
This line of code preserves the hyperlinks. Everything is working well:
`"qpdf.exe"  --linearize  "qpdf-test-hyperlinks.pdf" --pages  "qpdf-test-hyperlinks.pdf" 1-4   -- "chapter_01.pdf"`

Now I want to add a cover before each chapter:
`"qpdf.exe"  "cover.pdf " --linearize --pages . "qpdf-test-hyperlinks.pdf" 1-4   -- "chapter_01.pdf"`
The hyperlinks within the chapter no longer work.
The problem: The cover still contains metadata such as title and author. I would like to get that too
Attached are sample files
[cover.pdf](https://github.com/qpdf/qpdf/files/11983619/cover.pdf)
[qpdf-test-hyperlinks.pdf](https://github.com/qpdf/qpdf/files/11983620/qpdf-test-hyperlinks.pdf)

Is there a way to keep these hyperlinks alive?

Best regards.
Christoph

**Comment by jberkenbilt** [comment]
This is a known and long-standing issue and is related to a broader issue. There are many open issues related to this, all tagged with the `pages` label. Fixing this is next up on my qpdf to-do list. It will be a significant effort, but I'm hoping to get a large chunk of time to work on it sometime in the next few months.

**Comment by chrisrex** [comment]
Thank you for your report. I will wait with patience and look forward to the result.
Best regards
Christoph

**Comment by m-holger** [comment]
Hyperlinks are preserved for the main input file. If you make `qpdf-test-hyperlinks.pdf' the main file:
```
qpdf qpdf-test-hyperlinks.pdf --linearize --pages cover.pdf . 1-4 -- chapter1.pdf
```
the hyperlinks to the figures will be preserved.

**Comment by chrisrex** [comment]
Thank you m-holger,

it almost works. When opening chapter_01, page 1 (cover) is not opened, only the second page.

**Comment by m-holger** [comment]
Try opening the file, navigating back one page to the cover page and than exiting the viewer. When you reopen the file, hopefully you will land on the cover page.

**Comment by chrisrex** [comment]
Thanks. Acrobat does not remember the setting when exiting.
In the document settings, "Open on page" says 2 of 5.

**Comment by m-holger** [comment]
Try

```
qpdf qpdf-test-hyperlinks.pdf  --pages cover.pdf . 1-4 -- temp.pdf
qpdf temp.pdf --linearize chapter1.pdf
```

**Comment by chrisrex** [comment]
Thank you for your patience. Unfortunately, the result remains unchanged.

**Comment by m-holger** [comment]
Which page do you land on when you open the non-linearized file `temp.pdf`?

**Comment by chrisrex** [comment]
temp.pdf is landing on page 2 after opening

**Comment by m-holger** [comment]
This is very odd. I had both files opened with Acrobat Reader on a Windows system. In both cases, Acrobat reader landed on the cover page.

**Comment by chrisrex** [comment]
yes, that's strange. Thank you anyway for testing. I am using Acrobat Pro.

**Comment by m-holger** [comment]
I will look into this further, but it may take me a few days.

**Comment by chrisrex** [comment]
Thank you very much

**Comment by m-holger** [comment]
@chrisrex 

Could you confirm that 11.6.0 resolves your issue

**Comment by chrisrex** [comment]
Hi,It doesn't work with version 11.6 yet. I think that's announced for future version                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Von: m-holgerGesendet: Montag, 4. September 2023 14:54An: qpdf/qpdfAntwort an: qpdf/qpdfCc: chrisrex; MentionBetreff: Re: [qpdf/qpdf] split with hyperlinks (Issue #1003)
@chrisrex
Could you confirm that 11.6.0 resolves your issue

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/qpdf/qpdf/issues/1003#issuecomment-1705224535",
"url": "https://github.com/qpdf/qpdf/issues/1003#issuecomment-1705224535",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by jberkenbilt** [comment]
Preserving hyperlinks for joined files won't work until the `pages` epic is worked. Hyperlinks only work from the "primary" file for overlay, underlay, and pages.

**Comment by jberkenbilt** [comment]
See earlier comments -- perhaps there was a second issue? There was a fix in 11.6 to address this issue, but I admittedly didn't study the fix or the issue to see what part of the issue was addressed. @m-holger can comment.

**Comment by m-holger** [comment]
Faulty fix - I am looking into it

**Comment by m-holger** [comment]
@chrisrex 

Sorry, my fault.  When I tested my fix I must have picked up one of the workarounds I suggested earlier by mistake. Unfortunately your issue is not resolved by the current release.

**Comment by jberkenbilt** [comment]
I think there may still be work to do here. I'm pretty sure there are cases not related to any existing bug where splitting will break hyperlinks, particularly with named destinations. I have the information here on my pages roadmap.

**Comment by jberkenbilt** [comment]
This issue is part of the qpdf pages epic. If you are interested in following, please see #1104.
